tfx/extensions/google_cloud_ai_platform/training_clients.py

```diff
@@ -228,6 +230,10 @@ def create_training_job(self, input_dict: Dict[str, List[types.Artifact]],
         {telemetry_utils.LABEL_TFX_EXECUTOR: executor_class_path}):
       job_labels = telemetry_utils.make_labels_dict()
 
+    custom_config = json_utils.loads(
+        exec_properties.get(standard_component_specs.CUSTOM_CONFIG_KEY))
+    job_labels.update(custom_config.get('labels', {}))
```

Comment:
for cutom_config level keys, you would want to add it like [this](https://github.com/tensorflow/tfx/blob/master/tfx/extensions/google_cloud_ai_platform/trainer/executor.py#L40), thus users can find the keys and use it. consider make it an optional param to pass in this function from executor.py

Reply:
Good point @1025KB, I have added the `LABELS_KEY` to accommodate this. Minor logic change to merge labels in `executor.py`. In the `CAIPJobClient` I simply pop the `labels` dictionary from the `training_inputs` and update the `job_labels` dictionary.